dask/highlevelgraph.py

```diff
@@ -131,7 +131,10 @@ def from_collections(cls, name, layer, dependencies=()):
                     with ignoring(AttributeError):
                         deps[name] |= set(collection.__dask_layers__())
                 else:
-                    key = id(graph)
+                    try:
+                        key = collection.__dask_layers__()[0]
```

Comment:
Maybe this just to assert that we're not accidentally dropping any information here?  If this situation does arise (other collections exist) at least we'll fail more loudly.

```suggestion
                        [key] = collection.__dask_layers__()
```

Reply:
Sure - done.